### PR TITLE
Nested custom tags in QuakeML

### PR DESCRIPTION
**Pull request by claudiodsf**
This PR adds support for nested custom tags in QuakeML.

The page on ObsPy tutorial describing custom tags is updated accordingly.

Todo:
   - [x] test
   - [x] changelog

### CODE REVIEW

**Comment by claudiodsf** [comment]
+DOCS please

**Comment by claudiodsf** [comment]
This is good to go from my point of view.
Please review and let me know if you prefer to squash all commits into one.

File: obspy/io/quakeml/tests/test_quakeml.py [hunk 1]
```diff
@@ -881,7 +881,18 @@ def test_write_with_extra_tags_and_read(self):
              'tX': {'value': UTCDateTime('2013-01-02T13:12:14.600000Z'),
                     'namespace': r'http://test.org/xmlns/0.1'},
              'dataid': {'namespace': r'http://anss.org/xmlns/catalog/0.1',
-                        'type': 'attribute', 'value': '00999999'}})
+                        'type': 'attribute', 'value': '00999999'},
+             # some nested tags :
+             'quantity': {'namespace': r"http://some-page.de/xmlns/1.0",
```
Comment (krischer) [inline obspy/io/quakeml/tests/test_quakeml.py]: Using `r"..."` is a bit strange here. Why is that done? But I see that it is also used in the other test cases so am I missing something?

**Comment by krischer** [comment]
Looks pretty good to me :-) You can squash if you want - otherwise I think your commits are separate enough and could stay as they are.

File: obspy/io/quakeml/tests/test_quakeml.py (hunk 1 above)
Comment (claudiodsf) [inline obspy/io/quakeml/tests/test_quakeml.py]: Really don't know. I just modified a code which was originally written by @megies, I guess.

**Comment by claudiodsf** [comment]
Rebased + squashed + force-pushed

File: misc/docs/source/tutorial/code_snippets/quakeml_custom_tags.rst [hunk 2]
```diff
@@ -4,12 +4,12 @@
...
 =============================================================================
 
-QuakeML allows use of custom elements in addition to the "usual" information
+QuakeML allows use of custom elements in addition to the 'usual' information
 defined by the QuakeML standard. It allows *a)* custom namespace attributes to
 QuakeML namespace tags and *b)* custom namespace subtags to QuakeML namespace
 elements.
-ObsPy can handle both basic (non-nested) custom tags in event type objects
-(*a)*) and custom attributes (*b)*) during input/output to/from QuakeML.
+ObsPy can handle both basic custom tags in event type objects (*a)*) and custom
```
Comment (QuLogic) [inline misc/docs/source/tutorial/code_snippets/quakeml_custom_tags.rst]: I think you can remove these extra parentheses around the letters.

Comment (QuLogic) [inline misc/docs/source/tutorial/code_snippets/quakeml_custom_tags.rst]: Oh, I get it; it's a back-reference. In that case, remove the interior parenthesis instead.

File: misc/docs/source/tutorial/code_snippets/quakeml_custom_tags.rst [hunk 3]
```diff
@@ -94,3 +99,62 @@ the respective Event type objects (in this example to the Catalog object) as
...
+    my_tag.value.my_nested_tag1 = AttribDict()
+    my_tag.value.my_nested_tag1.namespace = ns
+    my_tag.value.my_nested_tag1.value = 1.23E+10
+
+    my_tag.value.my_nested_tag2 = AttribDict()
+    my_tag.value.my_nested_tag2.namespace = ns
+    my_tag.value.my_nested_tag2.value = True
+
+    cat = Catalog()
+    cat.extra = AttribDict()
```
Comment (QuLogic) [inline misc/docs/source/tutorial/code_snippets/quakeml_custom_tags.rst]: You could also write `AttribDict(my_tag=my_tag)`; don't know if you want to mention that here.

Comment (claudiodsf) [inline misc/docs/source/tutorial/code_snippets/quakeml_custom_tags.rst]: I think I'll leave it like this. I find it easier to follow.

File: obspy/io/quakeml/core.py [hunk 4]
```diff
@@ -987,7 +988,13 @@ def _extra(self, element, obj):
             for el in element.iterfind("{%s}*" % ns):
                 # remove namespace from tag name
                 _, name = el.tag.split("}")
-                value = el.text
+                # check if element has childs (nested tags)
```
Comment (QuLogic) [inline obspy/io/quakeml/core.py]: children

**Comment by claudiodsf** [comment]
Good to go, besides that `r"..."` thing, in case @megies has comments.

**Comment by megies** [comment]
> Good to go, besides that r"..." thing, in case @megies has comments.

I think the `r'..'` doesn't do anything, no idea why I put it in the original code. There are no backslashes in the strings (and in URLs backslashes are not allowed anyway), so it has no effect. Might be good to drop it, to not puzzle people in the future.

**Comment by claudiodsf** [comment]
Fixed the `r'..'` stuff. Good to merge for me.

**Pull request merged**